specs/node_types.md

```diff
@@ -0,0 +1,105 @@
+Node Types
+===
+
+- [Node Parameters](#node-parameters)
```

Comment:
IMO, it would be good to summarize the different node types and their security in some kind of matrix or table and add it to a section called Overview.

Reply:
NVM, I see there is a `## Node Types` at the bottom which is great!